src/PureScript/Backend/Optimizer/Semantics.purs

```diff
@@ -265,17 +265,8 @@ instance Eval f => Eval (BackendSyntax f) where
       evalAccessor env (eval env lhs) accessor
     Update lhs updates ->
       evalUpdate env (eval env lhs) (map (eval env) <$> updates)
-    Branch branches def -> do
-      let conds1 = evalPair env <$> branches
-      case def of
-        Just def' ->
-          evalBranches env conds1 (Just (defer \_ -> eval env def'))
-        Nothing ->
-          case (unwrap env).branchTry of
-            Nothing ->
-              evalBranches env conds1 Nothing
-            Just (Tuple conds2 def') ->
-              evalBranches env (NonEmptyArray.appendArray conds1 conds2) def'
+    Branch branches def ->
+      evalBranches env (evalPair env <$> branches) (Just (defer \_ -> eval env def))
```

Comment:
I think some work needs to be done here to remove the Maybe in evalBranches. This compiles, but it leaves a bunch of dead code in the evaluator.